go.mod

```diff
@@ -1,12 +1,12 @@
 module github.com/OctopusDeploy/terraform-provider-octopusdeploy
 
-go 1.22.0
+go 1.23.0
```

Comment:
I'm not sure what conventions are updating go versions, is this generally safe?

Reply:
generally safe, sometimes go will deprecate stuff but have time to replace it 👍